src/main/java/net/tiagofar78/prisonescape/game/prisonbuilding/SoundDetector.java

```diff
@@ -124,13 +130,46 @@ private void deleteFromWorld() {
         location.getBlock().setType(Material.AIR);
     }
 
+    private int calculateValue() {
+        return isDetectingSound() ? 1 : 0;
+    }
+
     private void updateValue() {
-        int value = isDetectingSound() ? MAX_VALUE : MIN_VALUE;
+        updateScoreboards(calculateValue());
+    }
+
+    private double noiseFunction(int x, int value) {
+        double intensity = 0.1;
+        return value + (_random.nextDouble() * 2 - 1) * intensity;
+    }
+
+    private void updateScoreboards(int value) {
+        _updateId++;
+        updateScoreboards(0, value, _updateId);
+    }
+
+    private void updateScoreboards(int x, int value, int updateId) {
+        PrisonEscapeGame game = GameManager.getGame();
+        if (game == null) {
+            return;
+        }
 
-        List<Guard> guards = GameManager.getGame().getGuardsTeam().getMembers();
+        List<Guard> guards = game.getGuardsTeam().getMembers();
         for (Guard guard : guards) {
-            guard.updateSoundDetectorValue(_index, value);
+            guard.updateSoundDetectorValue(_index, noiseFunction(x, value));
```

Comment:
I would rename noiseFunction to maybe like addRandomNoise().
Just so it's clearer what it does